src/clients/python/experimental_api_v2/examples/simple_http_v2_shm_string_client.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import argparse
+import numpy as np
+import sys
+from builtins import range
+import tritonhttpclient.core as httpclient
+import tritonsharedmemoryutils.shared_memory as shm
+import tritonhttpclient.utils as utils
+
+FLAGS = None
+
+if __name__ == '__main__':
```

Comment:
I think we are having too many v2 clients. We don't need a client for every combinations of different features, I think most of the clients were introduced as sanity check before v2 test transition and we should reducing them to minimal set of examples for showing how to use each feature as we have more and more test coverage on V2.

For example, what this client is testing should already be covered in `L0_infer_xxx_shared_memory`. And the features shown here is the combination of what are shown in `simple_http_v2_shm_client.py` and `simple_http_v2_string_infer_client.py`

Reply:
The reason this client is necessary is that the serialization of string data for the shm case happens outside the client library and in that case for string I/O hits different parts of the codebase

Reply:
Good point. And you should add a comment in the place you do the extra serialization / deserialization, about why the extra steps are needed.

Reply:
Done